dashboard/app/entities.go

```diff
@@ -514,3 +517,64 @@ func timeDate(t time.Time) int {
 	year, month, day := t.Date()
 	return year*10000 + int(month)*100 + day
 }
+
+type RecipientType int
+
+const (
+	To RecipientType = iota
+)
+
+func (t RecipientType) String() string {
+	return [...]string{"To", "Cc"}[t]
+}
+
+type RecipientInfo struct {
+	Address mail.Address
+	Type    RecipientType
+}
+
+type Recipients []RecipientInfo
+
+func (r Recipients) Len() int           { return len(r) }
+func (r Recipients) Less(i, j int) bool { return r[i].Address.Address < r[j].Address.Address }
+func (r Recipients) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
+
+func (r Recipients) String() string {
+	return strings.Join(r.StringSlice(), "|")
+}
+
+func (r Recipients) StringSlice() []string {
+	slice := []string{}
+	for _, addr := range r {
+		slice = append(slice, addr.Address.Address+"\n"+addr.Type.String())
+	}
+	return slice
+}
+
+func SplitEmails(emails []string) []string {
```

Comment:
Now I see the problem with this approach: we've lost type checking.
This function converts []string to []string. But these 2 have different meaning/contents. Trying to send emails to the wrong []string will lead to runtime bugs.
You sprinkled some SplitEmails in the code. How do we know you added them in all needed places? How do we know you did not added in the wrong places? How future reviewers of future changes by other people will know that SplitEmails is used in all right places? Most likely they won't think that there is such a concern.

I wonder if we could add a custom type for the other `string`:
```
type Something string
```
and if datastore will handle this. If yes, then since Go does not have implicit conversions, it will solve the problem.